mrjob/dataproc.py

```diff
@@ -0,0 +1,873 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import hashlib
+import json
+import logging
+import os
+import os.path
+import pipes
+import time
+import re
+
+try:
+    from oauth2client.client import GoogleCredentials
+    from googleapiclient import discovery
+except ImportError:
+    # don't require googleapiclient; MRJobs don't actually need it when running
+    # inside hadoop streaming
+    GoogleCredentials = None
+    discovery = None
+
+try:
+    import filechunkio
+except ImportError:
+    # that's cool; filechunkio is only for multipart uploading
+    filechunkio = None
+
+from mrjob.conf import combine_dicts
+from mrjob.conf import combine_lists
+from mrjob.conf import combine_paths
+
+from mrjob.fs.composite import CompositeFilesystem
+from mrjob.fs.local import LocalFilesystem
+
+from mrjob.fs.gcs import GCSFilesystem
+
+from mrjob.logs.counters import _pick_counters
+from mrjob.fs.gcs import is_gcs_uri
+from mrjob.parse import is_uri
+
+from mrjob.runner import MRJobRunner
+from mrjob.runner import RunnerOptionStore
+from mrjob.setup import BootstrapWorkingDirManager
+from mrjob.setup import UploadDirManager
+from mrjob.setup import parse_setup_cmd
+from mrjob.step import StepFailedException
+from mrjob.util import cmd_line
+from mrjob.util import random_identifier
+
+log = logging.getLogger(__name__)
+
+_DEFAULT_GCP_REGION = 'global'
+_HADOOP_STREAMING_JAR_URI = 'file:///usr/lib/hadoop-mapreduce/hadoop-streaming.jar'
+
```

Comment:
You're going to need a way to specify your machine image, and learn what version of Hadoop it's using.